test/functional/apps/discover/_discover.js

```diff
@@ -11,120 +11,87 @@ import {
-    var fromTimeString = 'September 19th 2015, 06:31:44.000';
-    var toTimeString = 'September 23rd 2015, 18:31:44.000';
-
-    bdd.it('should show correct time range string by timepicker', function () {
-      var expectedTimeRangeString = fromTimeString + ' to ' + toTimeString;
-      return PageObjects.discover.getTimespanText()
-      .then(function (actualTimeString) {
-        expect(actualTimeString).to.be(expectedTimeRangeString);
-      });
+
+    bdd.it('should show correct time range string by timepicker', async function () {
+      var actualTimeString = await PageObjects.discover.getTimespanText();
+
+      expect(actualTimeString).to.be(`${fromTimeString} to ${toTimeString}`);
     });
 
-    bdd.it('save query should show toast message and display query name', function () {
-      var expectedSavedQueryMessage = 'Discover: Saved Data Source "' + queryName1 + '"';
-      return PageObjects.discover.saveSearch(queryName1)
-      .then(function () {
-        return PageObjects.header.getToastMessage();
-      })
-      .then(function (toastMessage) {
-        PageObjects.common.saveScreenshot('Discover-save-query-toast');
-        expect(toastMessage).to.be(expectedSavedQueryMessage);
-      })
-      .then(function () {
-        return PageObjects.header.waitForToastMessageGone();
-      })
-      .then(function () {
-        return PageObjects.discover.getCurrentQueryName();
-      })
-      .then(function (actualQueryNameString) {
-        expect(actualQueryNameString).to.be(queryName1);
-      });
+    bdd.it('save query should show toast message and display query name', async function () {
+      await PageObjects.discover.saveSearch(queryName1);
+      var toastMessage = await PageObjects.header.getToastMessage();
+      await PageObjects.common.saveScreenshot('Discover-save-query-toast');
+
+      expect(toastMessage).to.be(`Discover: Saved Data Source "${queryName1}"`);
+
+      await PageObjects.header.waitForToastMessageGone();
+      var actualQueryNameString = await PageObjects.discover.getCurrentQueryName();
+
+      expect(actualQueryNameString).to.be(queryName1);
     });
 
-    bdd.it('load query should show query name', function () {
-      return PageObjects.discover.loadSavedSearch(queryName1)
-      .then(function () {
-        return PageObjects.common.sleep(3000);
-      })
-      .then(function () {
-        return PageObjects.discover.getCurrentQueryName();
-      })
-      .then(function (actualQueryNameString) {
-        PageObjects.common.saveScreenshot('Discover-load-query');
-        expect(actualQueryNameString).to.be(queryName1);
+    bdd.it('load query should show query name', async function () {
+      await PageObjects.discover.loadSavedSearch(queryName1);
+
+      await PageObjects.common.saveScreenshot('Discover-load-query');
+      await PageObjects.common.try(async function() {
+        expect(await PageObjects.discover.getCurrentQueryName()).to.be(queryName1);
```

Comment:
Waiting for the spinner to be shown might lead to false positives too if it vanishes quickly (as it should if the app performs well).